core/execution/future/market.go

```diff
@@ -1023,6 +1036,24 @@ func (m *Market) BlockEnd(ctx context.Context) {
 		m.indexPriceCalculator.CalculateBookMarkPriceAtTimeT(m.tradableInstrument.MarginCalculator.ScalingFactors.InitialMargin, m.mkt.LinearSlippageFactor, m.risk.GetRiskFactors().Short, m.risk.GetRiskFactors().Long, t.UnixNano(), m.matching)
 	}
 
+	// if we do have a separate configuration for the index price and we have a new index price we push it to the perp
+	if m.indexPriceCalculator != nil && (m.nextIndexPriceCalc.IsZero() ||
+		!m.nextIndexPriceCalc.After(t) &&
+			!m.as.InAuction()) { // TODO @zohar do we want to update the mark price and or MTM during auctions?
+		prevIndexPrice := m.indexPriceCalculator.price
+		m.indexPriceCalculator.CalculateMarkPrice(
+			t.UnixNano(),
+			m.matching,
+			m.mtmDelta,
+			m.tradableInstrument.MarginCalculator.ScalingFactors.InitialMargin, m.mkt.LinearSlippageFactor, m.risk.GetRiskFactors().Short, m.risk.GetRiskFactors().Long)
+		m.nextIndexPriceCalc = t.Add(m.indexConfigFrequency)
+		if (prevIndexPrice == nil || !m.indexPriceCalculator.price.EQ(prevIndexPrice) || m.settlement.HasTraded()) &&
+			!m.getCurrentIndexPrice().IsZero() {
+			m.tradableInstrument.Instrument.Product.SubmitDataPoint(ctx, m.getCurrentIndexPrice(), m.timeService.GetTimeNow().UnixNano())
+		}
+	}
+
+	// if it's time for mtm, let's do it
 	if m.nextMTM.IsZero() ||
 		!m.nextMTM.After(t) &&
 			!m.as.InAuction() { // TODO @zohar do we want to update the mark price and or MTM during auctions?
```

Comment:
in response to the comment: No for MTM, not sure about mark price

Reply:
yeah I don't think the goal is to MTM during auctions but I think there're a requirement to update the mark price during auctions. @davidsiska-vega

Reply:
don’t do mark price updates (even if they come from external oracle) during auctions. We do a MTM update upon leaving the auction with the latest available value.